src/lib/comms/sol-lwm2m.c

```diff
@@ -4730,9 +5037,18 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
     struct resource_ctx *res_ctx;
     struct sol_lwm2m_resource res[3];
     int r;
+    static bool setup_access_control = true;
```

Comment:
Two clients? I didn't get it.

`setup_access_control_object_instances();` should be called only once, no matter what. It instantiates one Access Control Object Instance for each Object supported by the Client. This doesn't change.

The only case in which `setup_access_control_object_instances();` needs to be called again is if a Bootstrap Server sends `Delete "/"`, since that will delete these Access Control Instances as well, and then they will need to be recreated.

This function (`setup_access_control_object_instances();`) is responsible for the access rights related to each Object, not Object Instance.